.github/actions/install-wdk/action.yml

```diff
@@ -136,21 +136,22 @@ runs:
 
           Write-Host "Using Winget to install WDK and SDK version $inputVersion..."
 
-          # Install Windows SDK 10.0.22621 explicitly when WDK 10.0.22621 is requested
-          if ($sdkVersion -eq "10.0.22621") {
-            Write-Host "Installing 10.0.22621 SDK ..."
-            $sdk = "Microsoft.WindowsSDK.$sdkVersion"
-            Install-WinGetPackage -Id $sdk -Source winget -MatchOption Equals -Mode Silent -Force -Override '/q /features +'
+          $sdk = "Microsoft.WindowsSDK.$sdkVersion"
+          $wdk = "Microsoft.WindowsWDK.$inputVersion"
+          Write-Host "Installing $sdk..."
+          Install-WinGetPackage -Id $sdk -Source winget -MatchOption Equals -Mode Silent -Force -Override '/q /features +'
+          $installSuccess = $?
+          if (-not $installSuccess) {
+            Write-Error "$sdk installation failed. Last exit code: $LASTEXITCODE and success status: $installSuccess"
+            exit 1
           }
 
-          # Install Windows WDK
-          $wdk = "Microsoft.WindowsWDK.$inputVersion"
-          if ((Get-WinGetPackage -Id $wdk -Source winget -MatchOption Equals).Id -eq '$wdk') {
-            Write-Host "$wdk is already installed. Attempting to update..."
-            Update-WinGetPackage -Id $wdk -Source winget -MatchOption Equals -Mode Silent -Force
-          } else {
-            Write-Host "Installing $wdk..."
-            Install-WinGetPackage -Id $wdk -Source winget -MatchOption Equals -Mode Silent -Force
+          Write-Host "Installing $wdk..."
+          Install-WinGetPackage -Id $wdk -Source winget -MatchOption Equals -Mode Silent -Force
+          $installSuccess = $?
```

Comment:
Nit: you don't really need this `$installSuccess` variable because `$?` is a well known variable in PowerShell and people known what it means.

Reply:
That's true, but the main reason for adding this variable is that we needed the status code in two places: once for the `if` condition and again to print it using the `Write-Error` log. I'm being extra cautious in case someone adds a new line between the `if` check and the `Write-Error` statement, as `$?` would then reflect the status of the new line instead of `Install-WinGetPackage`. This could be easy to overlook and difficult to catch.